### DISCUSSION TITLE
SvelteKit Support

### DISCUSSION THREAD

**Discussion by acurrieclark** [cawa-93/vite-electron-builder#912] [Forks]
I have made an attempt at getting this working with SvelteKit. The fork can be found here: https://github.com/acurrieclark/vite-electron-builder

There are a few caveats, predominantly that I can't find a way to maintain the `packages/renderer` structure from the main repo. This is more of a SvelteKit issue than an electron one, and everything seems to work as expected.

Would appreciate people's feedback. happy building!

**Comment by van100j** [comment]
Yeah, SvelteKit seems very sensitive when its root directory differs from the current working directory. 

Created [SvelteKit fork](https://github.com/van100j/sveltekit-vite-electron-builder) and kept it in `packages/renderer`.

The key was to make sure we run [Vite's `createServer` with `packages/renderer` being the `cwd()`](https://github.com/van100j/sveltekit-vite-electron-builder/blob/main/scripts/watch.mjs#L9). Also needed to update Vite's `config` for [main package watcher](https://github.com/van100j/sveltekit-vite-electron-builder/blob/main/scripts/watch.mjs#L34) and [preload package watcher](https://github.com/van100j/sveltekit-vite-electron-builder/blob/main/scripts/watch.mjs#L77).

**Reply by acurrieclark** [reply]
Ah, well done. Will take a look. Appreciate the feedback
